### PR TITLE
Hooks

### PR DESCRIPTION
**Pull request by Yehor-K**

### CODE REVIEW

File: Yehor/reactNetflix/src/components/detailedMovie/DetailedMovie.tsx [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
-
-  const defimg = (e: React.SyntheticEvent<HTMLImageElement>) => {
-    const target = e.target as HTMLImageElement;
-    target.src = "https://media.comicbook.com/files/img/default-movie.png";
-  }
-
   return (
     <div className="dmovie">
-      <img src={movie.poster_path} alt={movie.title} onError={defimg} className="dmovie_img"/>
+      <ImageFallBack
```
Comment (brusni4ka) [inline Yehor/reactNetflix/src/components/detailedMovie/DetailedMovie.tsx:13]: 👍

File: Yehor/reactNetflix/src/components/errorBoundary/ErrorBoundary.tsx [hunk 2]
```diff
@@ -1,31 +1,35 @@
...
-        )
-      }
-      return this.props.children
-    }
+type IErrorBoundaryProps = {
+  // children: JSX.Element; ???????
+  children: any;
+}
+
+const ErrorBoundary = (props: IErrorBoundaryProps) => {
```
Comment (brusni4ka) [inline Yehor/reactNetflix/src/components/errorBoundary/ErrorBoundary.tsx]: Nice that you tried to create functional ErrorBoundary component. But better to use class for this case for now. https://reactjs.org/docs/hooks-faq.html#do-hooks-cover-all-use-cases-for-classes

File: Yehor/reactNetflix/src/components/search-panel/SearchPanel.tsx [hunk 3]
```diff
@@ -23,125 +19,91 @@ enum SearchBy {
...
 
-class SearchPanel extends React.Component<IhandleSearchChangeProps, ISearchPanelState> {
-  state: ISearchPanelState = {
-    value: "",
-    searchBy: SearchBy.Title,
+const SearchPanel = (props: IhandleSearchChangeProps) => {
+  const [value, setValue] = useState("");
+  const [searchBy, setSearchBy] = useState(SearchBy.Title);
+
+  useEffect(() => {
```
Comment (brusni4ka) [inline Yehor/reactNetflix/src/components/search-panel/SearchPanel.tsx:26]: You don't need this useEffect. The useEffect that you have on 35 will do the same and also will be called on mount

File: Yehor/reactNetflix/src/components/search-panel/SearchPanel.tsx [hunk 4]
```diff
@@ -23,125 +19,91 @@ enum SearchBy {
...
+    const querySrch = parse(props.location.search) as {
+      searchBy: string;
+      search: string;
+    };
+    const { searchBy, search } = querySrch;
+    checkSetSearchSearchBy(searchBy, search);
+  }, [props.location]);
+
+  const checkSetSearchSearchBy = (
```
Comment (brusni4ka) [inline Yehor/reactNetflix/src/components/search-panel/SearchPanel.tsx]: I would suggest to change the name of the function. Looks really confusing to me. you don't really check, you set it anyway.

File: Yehor/reactNetflix/src/components/sort-panel/SortPanel.tsx [hunk 5]
```diff
@@ -17,75 +17,51 @@ enum SortBy {
   Rating = "vote_average",
 }
 
-class SortPanel extends React.Component<ISortPanelProps, ISortPanelState> {
-  state = {
-    sortBy: SortBy.Date,
+const SortPanel = (props: ISortPanelProps) => {
+  const [sortBy, setSortBy] = useState(SortBy.Date);
+
+  useEffect(() => {
```
Comment (brusni4ka) [inline Yehor/reactNetflix/src/components/sort-panel/SortPanel.tsx:23]: you don't need this useEffect.

File: Yehor/reactNetflix/src/pages/detailedPage/DetailedPage.tsx [hunk 6]
```diff
@@ -1,90 +1,108 @@
...
+        )}
+      </div>
+      <div className="main-container">
         <MoviesList movies={movies} />
+        <div className="pagination">
+          <button onClick={props.dataOffsetDecrement}>Back</button>
+          <span className="pages">
+            {props.offset ? props.offset / 9 + 1 : 1}
+          </span>
+          <button onClick={props.dataOffsetIncrement}>Next</button>
```
Comment (brusni4ka) [inline Yehor/reactNetflix/src/pages/detailedPage/DetailedPage.tsx]: Movies are not endless, would be grate if you wrapper your button in some condition. Hide button if there is no more movie left to fetch

File: Yehor/reactNetflix/src/pages/homePage/HomePage.tsx [hunk 7]
```diff
@@ -8,125 +8,117 @@ import Footer from "../../components/footer";
...
   dataOffsetDecrement,
-  receivedData,
-  ILoadData,
-} from "../../redux/actions/moviesActions";
+} from "../../redux/reducers/reducerMovies";
 import { connect, ConnectedProps } from "react-redux";
-import IReduxState from "../../interfaces/IReduxState"
+import IReduxState from "../../interfaces/IReduxState";
 
 const mapStateToProps = (state: IReduxState) => {
```
Comment (brusni4ka) [inline Yehor/reactNetflix/src/pages/homePage/HomePage.tsx]: you can replace mapStateToProps with useSelector from redux/toolkit

File: Yehor/reactNetflix/src/pages/homePage/HomePage.tsx [hunk 8]
```diff
@@ -8,125 +8,117 @@ import Footer from "../../components/footer";
...
 
 const connector = connect(mapStateToProps, mapDispatchToProps);
 type PropsFromRedux = ConnectedProps<typeof connector> & RouteComponentProps;
 
-class HomePage extends React.Component<PropsFromRedux, {}> {
-  componentDidMount() {
-    const queryUrl = parse(this.props.location.search) as {
+const HomePage = (props: PropsFromRedux) => {
+  useEffect(() => {
+    const queryUrl = parse(props.location.search) as {
```
Comment (brusni4ka) [inline Yehor/reactNetflix/src/pages/homePage/HomePage.tsx:26]: check your network. It seems like you have additional request when you apply some filters. You need to add `history.action !== 'PUSH'` check here

File: Yehor/reactNetflix/src/redux/sagas/moviesSaga.ts [hunk 9]
```diff
@@ -1,95 +1,88 @@
...
+  search?: string,
+  offset?: number
+): Promise<IMovie[]> => {
+  const limit = 9;
+  const url =
+    searchBy && search
+      ? `https://reactjs-cdp.herokuapp.com/movies?sortBy=
+  ${sortBy ? sortBy : "release_data"}
+  &search=${search}&searchBy=${searchBy}&offset=${offset}&limit=${limit}`
+      : `https://reactjs-cdp.herokuapp.com/movies?sortBy=release_data&sortOrder=desc&offset=${offset}&limit=${limit}`;
```
Comment (brusni4ka) [inline Yehor/reactNetflix/src/redux/sagas/moviesSaga.ts]: Ну такое....  Still don't like. Too many letters, hard to read. Think of better solution

File: Yehor/reactNetflix/src/redux/sagas/moviesSaga.ts [hunk 10]
```diff
@@ -1,95 +1,88 @@
...
 
-export const fetchCurrentMovieApi = async (id: string): Promise<void> => {
-  const getMovies = await fetch(
-    `http://reactjs-cdp.herokuapp.com/movies/${id}`
-  );
-  const movie = await getMovies.json();
-  console.log(movie);
-  return movie;
-};
+interface ILoadDataAction {
```
Comment (brusni4ka) [inline Yehor/reactNetflix/src/redux/sagas/moviesSaga.ts]: don't like that you had to create an interface just to use it in saga. Seems redundant to me. My proposal is to use this instead:
ReturnType<typeof loadData >

File: Yehor/reactNetflix/src/pages/homePage/HomePage.tsx (hunk 8 above)
Comment (Yehor-K) [reply Yehor/reactNetflix/src/pages/homePage/HomePage.tsx:26]: I have checked. Everything is fine, I don't have additional request

**Pull request merged**